### PR TITLE
OBSDOCS-205 - CLF audit log filtering

### PR DESCRIPTION
**Pull request by libander**
Version(s): 4.12+

Issue: [OBSDOCS-205](https://issues.redhat.com//browse/OBSDOCS-205)

Link to docs preview: https://70608--ocpdocs-pr.netlify.app/openshift-enterprise/latest/logging/log_collection_forwarding/configuring-log-forwarding#logging_audit_filtering_configuring-log-forwarding

QE review: @anpingli / @kabirbhartiRH / @QiaolingTang 

- [x] QE has approved this change.

Additional information:
SME review: @alanconway / @jcantrill

### CODE REVIEW

File: modules/logging-audit-log-filtering.adoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+The API Audit filter provides a solution to this problem by enabling the exclusion of non-essential events and the reduction of event size, facilitating a more manageable audit trail.
+
+An audit policy event contains meta-data describing who made the request. It can also include the full body of the API request, and the response that was sent. Rules are checked in order, checking stops at the first matching rule. The `level` of an audit rule determines how much data is included in the event:
+
+* None: The event is dropped.
+* Metadata: Only the audit metadata is included, request and response bodies are removed.
+* Request: Audit metadata and the request body are included, the response body is removed.
+* RequestResponse: All data is included: metadata, request body and response body. The response body can be very large. For example, `oc list -A pods` generates a response body containing the YAML description of every pod in the cluster.
```
Comment (QiaolingTang) [inline modules/logging-audit-log-filtering.adoc:16]: Seems `oc list` is not a valid command, I got:

```
$ oc list
error: unknown command "list" for "oc"

Did you mean this?
	get
	wait
```

**Comment by snarayan-redhat** [comment]
@libander I would suggest submitting this for peer review once you get a QE approval.

**Comment by JoeAldinger** [comment]
@libander do you mind adding the assembly file so the preview works, please?

File: modules/logging-audit-log-filtering.adoc (hunk 1 above)
Comment (libander) [reply modules/logging-audit-log-filtering.adoc]: @QiaolingTang - fixed, thank you!

**Comment by QiaolingTang** [comment]
LGTM.

**Review by snarayan-redhat — COMMENTED** [review]
Good effort. I have left few suggestions which might want to consider.

**Comment by snarayan-redhat** [comment]
For now, I am not adding a `peer-review-done` label since there are some comments that need to be addressed. You can request for a review once ready by adding `peer-review-needed` label again.

**Review by jcantrill — COMMENTED** [review]
/lgtm

File: modules/logging-audit-log-filtering.adoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
+// Module included in the following assemblies:
+//
+//
```
[the remark this answers is not in the document]
Comment (libander) [reply modules/logging-audit-log-filtering.adoc:3]: Good catch, thanks!

**Review by dfitzmau — COMMENTED** [review]
Hi @libander . Nice work on the contnet. I added some minor comments but the YAML file might indicate a bigger issue for customers that copy then example verbatim. I see some red text highlighting in the preview that might indicate an issue.

**Comment by libander** [comment]
> Hi @libander . Nice work on the contnet. I added some minor comments but the YAML file might indicate a bigger issue for customers that copy then example verbatim. I see some red text highlighting in the preview that might indicate an issue.

Thanks! Added admonitions to those fields, and corrected the admonition. That example is not intended to be copied as it isn't a recommended policy. :)

File: modules/logging-audit-log-filtering.adoc [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+To disable these defaults, either end your rules list with a rule that has only a `level` field or add an empty rule.
+
+Omit Response Codes:: A list of integer status codes to omit. You can drop events based on the HTTP status code in the response by using the `OmitResponseCodes` field, a list of HTTP status code for which no events are created. The default value is `[404, 409, 422, 429]`. If the value is an empty list, `[]`, then no status codes are omitted.
+
+The `ClusterLogForwarder` CR audit policy acts in addition to the {product-title} audit policy. The `ClusterLogForwarder` CR audit filter changes what the log collector forwards, and provides the ability to filter by verb, user, group, namespace, or resource. You can create multiple filters to send different summaries of the same audit stream to different places. For example, you can send a detailed stream to the local cluster log store, and a less detailed stream to a remote site.
+
+[NOTE]
+====
+The example provided is intended to illustrate of the range of rules possible in an audit policy and is not a recommended configuration.
```
Comment (mburke5678) [inline modules/logging-audit-log-filtering.adoc:31]:
```suggestion
The example provided is intended to illustrate the range of rules possible in an audit policy and is not a recommended configuration.
```

File: modules/logging-audit-log-filtering.adoc [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogForwarder
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  pipelines:
+    - name: my-pipeline
+      inputRefs: #<1>
+      filterRefs: #<2>
```
Comment (mburke5678) [inline modules/logging-audit-log-filtering.adoc:47]: Would it make sense to add some examples to the `inputRefs` and `filterRefs` parameters, especially where the code example is purely illustrative. It took me several minutes to figure it out (assuming I did):

```
  pipelines:
    - name: my-pipeline
      inputRefs: audit #<1>
      filterRefs: my-policy #<2>
```

**Comment by mburke5678** [comment]
@libander I have a couple of nits I was hoping you could look at before we merge.

**Review by mburke5678 — APPROVED** [review]

**Pull request merged**